src/libs/antares/study/parts/short-term-storage/series.cpp

```diff
@@ -180,4 +180,120 @@ bool Series::validateLowerRuleCurve() const
     return checkVectBetweenZeroOne(maxInjectionModulation, "lower rule curve");
 }
 
+bool Series::validateInflowsSums(bool simplexIsWeek, unsigned int cycleDuration,
+        unsigned int simuFirstHour, unsigned int simuLastHour) const
+{
+    // allows to loop on hours even if simulation starts in november and ends in january or later
```

Comment:
Please don't loop inside this function.

Add a function for 1 week here, call it in `Area::checkWeeklyData`